azurerm/data_source_storage_account_sas.go

```diff
@@ -0,0 +1,390 @@
+package azurerm
+
+import "github.com/hashicorp/terraform/helper/schema"
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+)
+
+const (
+	connStringAccountKeyKey  = "AccountKey"
+	connStringAccountNameKey = "AccountName"
+	sasSignedVersion         = "2017-07-29"
+)
+
+// This is an ACCOUNT SAS : https://docs.microsoft.com/en-us/rest/api/storageservices/Constructing-an-Account-SAS
+// not Service SAS
+func dataSourceArmStorageAccountSharedAccessSignature() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmStorageAccountSasRead,
+
+		Schema: map[string]*schema.Schema{
+			"connection_string": {
+				Type:      schema.TypeString,
+				Required:  true,
+				ForceNew:  true,
+				Sensitive: true,
+			},
+
+			"https_only": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+				ForceNew: true,
+			},
+
+			"resource_types": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"service": {
```

Comment:
Would it be possible to default all of these (the `resource_types`, `permissions`, and `services` parameters) to `false` and make them optional?

Reply:
@tombuildsstuff Agree with all of the above and will incorporate those changes.
@phekmat I debated that myself and came to the conclusion to be explicitly verbose here since there are security vs function issues here. 

If we default all to false, then there's an implicit no-access (all broken) case. 

IHMO it's better here to be verbose and demand that folks specify exactly what they want... 

This avoids an oops by granting too much access and an oops by granting no access both of which, IMHO, are a worse UX than some extra typing.